pyazul/validate.py

```diff
@@ -1,12 +1,60 @@
+""" Validation module
+"""
+from . import exceptions
 from . import utils
 
 
+def check_not_empty(value):
+    if not value:
+        raise exceptions.UnexpectedEmptyValue
+    return True
+
+
+def check_minimum_length(length):
+    def inner(value):
+        value_len = len(value)
+        if value_len < length:
+            raise exceptions.MinimumLengthNotReached(
+                f'{value}: current length {value_len}'
+                f'expected {length}'
+            )
+        return True
+    return inner
+
+
+def check_exists_in(data):
+    def inner(key):
+        if key not in data:
+            raise exceptions.RequiredParameterNotFound(key)
+        return True
+    return inner
+
+
+def check_type(type):
+    def inner(value):
+        if not isinstance(value, type):
+            raise exceptions.UnsuportedType
+        return True
+    return inner
+
+
+def validate(data, validation_rule_sets):
+
+    for rule_set in validation_rule_sets:
+
+        value = data[rule_set]
+        rule_set = validation_rule_sets[rule_set]
+
+        for rule in rule_set:
+            rule(value)
```

Comment:
This may have happened because this validate method is generic for both key and value validations. Since you're only passing value and not the key, a method like 'check_exists_in' which is expecting a key just doesn't work.